File: 198. House Robber/rob.java
public class Solution {
    public int rob(int[] nums) {
        if (nums.length == 0){
            return 0;
        }
        else if (nums.length == 1){
            return nums[0];
        }
        int[] money = new int[nums.length];
        money[0] = nums[0];
        money[1] = Math.max(nums[0], nums[1]);
        for (int i = 2; i < nums.length; i++){
            money[i] = Math.max(money[i-2]+nums[i], money[i-1]);
            // System.out.println("money[" + i + "] = " + money[i]);
        }
        // printArr(money);
        return money[nums.length-1];
    }
    
    public void printArr(int[] nums){
		for (int i = 0; i < nums.length; i++) {
			System.out.print("'"+nums[i]+"',");
		}
	}
}

File: 198. House Robber/rob.py
class Solution(object):
    def rob(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        # max(f(n) + f(n-2), f(n-1))
        last2 = last1 = cur = 0
        for i in range(len(nums)):
            cur = max(nums[i] + last2, last1)
            last1, last2 = cur, last1
        return cur
            